src/vfs2.c

```diff
@@ -403,6 +404,37 @@ struct shm {
 static_assert(sizeof(struct shm_region) == VFS2_WAL_INDEX_REGION_SIZE,
 	      "shm regions have the expected size");
 
+/**
+ * Allocate a new shm region at the next-highest index.
+ *
+ * Returns a pointer to the new region, or NULL if allocation failed.
+ * In the latter case the shm is unchanged.
+ */
+static struct shm_region *shm_grow(struct shm *shm)
+{
+	int index = shm->num_regions;
```

Comment:
Nit: you can move this closer to usage.

Reply:
I'm not sure about this---it's used by shm_init which is just two functions down (and the function in the middle, write_basic_hdr_cksums, is also used by shm_init). (I do think the overall order of functions in the file needs tidying up, just not sure how to achieve a quick improvement in this case.)